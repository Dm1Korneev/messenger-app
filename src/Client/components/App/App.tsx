import Box from '@mui/material/Box';
import CssBaseline from '@mui/material/CssBaseline';

import { useIsLoggedIn } from 'Client/hooks';

import { MessengerScreen } from '../MessengerScreen';
import { SignIn } from '../SignIn';

export const App = () => {
  const isLoggedIn = useIsLoggedIn();

  return (
    <Box display="flex" height="100vh">
      <CssBaseline />
      {isLoggedIn ? <MessengerScreen /> : <SignIn />}
    </Box>
  );
};
