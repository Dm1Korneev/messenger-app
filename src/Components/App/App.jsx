import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import Box from '@material-ui/core/Box';
import CssBaseline from '@material-ui/core/CssBaseline';

import MessengerScreen from 'Components/MessengerScreen';
import SignIn from 'Components/SignIn';

const App = ({
  isLoggedIn,
  loginFromStore,
}) => {
  useEffect(() => {
    loginFromStore();
  }, [loginFromStore]);

  let result;
  if (!isLoggedIn) {
    result = <SignIn />;
  } else {
    result = (
      <MessengerScreen />
    );
  }

  return (
    <Box height="100vh" display="flex">
      <CssBaseline />
      {result}
    </Box>
  );
};

App.propTypes = {
  loginFromStore: PropTypes.func.isRequired,
  isLoggedIn: PropTypes.bool.isRequired,
};

export default App;
