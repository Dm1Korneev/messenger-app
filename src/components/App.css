.App__content {
  max-width: 1010px;
  min-width: 300px;
  margin: 0 auto;
  padding: 1em;
  background: #fff;
  border-radius: 0.5em;

  height: 90vh;
  display: flex;
  flex-flow: column;
  justify-content: space-between;
}
