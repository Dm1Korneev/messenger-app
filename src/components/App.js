import React, { Component } from "react";

// components
import MessagesList from "./MessagesList";
import MessageInput from "./MessageInput";
import SignIn from "./SignIn";

// @material-ui
import withStyles from "@material-ui/core/styles/withStyles";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import { blue } from "@material-ui/core/colors";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import Button from "@material-ui/core/Button";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";

// common
import { getMessages, sendMessage } from "../common/messengerAPI";
import {
  isLoggedIn,
  getToken,
  currentUser,
  login,
  register,
  saveToken,
  logout
} from "../common/authentication";

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      messages: [],
      token: getToken(),
      currentUser: currentUser()
    };

    this.theme = createMuiTheme({
      palette: {
        primary: blue
      },
      typography: {
        useNextVariants: true
      }
    });
  }

  componentDidMount() {
    const { token } = this.state;
    if (isLoggedIn(token)) {
      this.loadMessages();
    }
  }

  loadMessages = () => {
    const { token } = this.state;

    getMessages(token, messages => this.setState({ messages: messages }));
  };

  onSendMessage = messageText => {
    const { token } = this.state;

    sendMessage(token, messageText);
    this.loadMessages();
  };

  handleClickLogout = () => {
    logout();
    this.setState({
      token: undefined
    });
  };

  onSignIn = ({ email, password, remember }) => {
    login(email, password, result => {
      const { token } = result;

      if (remember) {
        saveToken(token);
      }
      this.setState({
        token: token
      });
      this.loadMessages();
    });
  };

  onRegister = ({ email, password, name, remember }) => {
    register(email, password, name, result => {
      const { token } = result;

      if (remember) {
        saveToken(token);
      }
      this.setState({
        token: token
      });
      this.loadMessages();
    });
  };

  render() {
    const { classes } = this.props;
    const { messages, token } = this.state;
    const user = currentUser(token);

    let result;
    if (!isLoggedIn(token)) {
      result = <SignIn onSignIn={this.onSignIn} onRegister={this.onRegister} />;
    } else {
      result = (
        <>
          <AppBar position="static">
            <Toolbar variant="dense">
              <IconButton
                className={classes.menuButton}
                color="inherit"
                aria-label="Menu"
              >
                <MenuIcon />
              </IconButton>
              <Typography variant="h6" color="inherit" className={classes.grow}>
                {user.name}
              </Typography>
              <Button color="inherit" onClick={this.handleClickLogout}>
                logout
              </Button>
            </Toolbar>
          </AppBar>
          <Grid item xs={12} sm={11} md={9} lg={7} className={classes.grid}>
            <Paper className={classes.content}>
              <MessagesList messages={messages} />
              <MessageInput onSendMessage={this.onSendMessage} />
            </Paper>
          </Grid>
        </>
      );
    }

    return (
      <main className={classes.root}>
        <MuiThemeProvider theme={this.theme}>{result}</MuiThemeProvider>
      </main>
    );
  }
}

const styles = theme => ({
  root: {
    height: "100vh"
  },
  grid: {
    margin: "0 auto",
    height: "calc(100% - 48px)",
    padding: theme.spacing.unit * 2
  },
  content: {
    height: "100%",
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    paddingLeft: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingTop: theme.spacing.unit
  },
  grow: {
    flexGrow: 1
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  }
});

export default withStyles(styles)(App);
