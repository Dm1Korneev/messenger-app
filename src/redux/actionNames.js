export const GET_CHATS = 'GET_CHATS';
export const CHATS_ADD = 'CHATS_ADD';
export const CREATE_CHAT = 'CREATE_CHAT';
export const MODIFY_CHAT = 'MODIFY_CHAT';

export const GET_MESSAGES = 'GET_MESSAGES';
export const MESSAGES_ADD = 'MESSAGES_ADD';
export const SEND_MESSAGE = 'SEND_MESSAGE';

export const GET_USERS = 'GET_USERS';
export const USERS_ADD = 'USERS_ADD';
export const MODIFY_USER = 'MODIFY_USER';

export const STORE_CLEAR = 'STORE_CLEAR';

export const LOGIN = 'LOGIN';
export const REGISTER = 'REGISTER';
export const LOGIN_FROM_STORE = 'LOGIN_FROM_STORE';
export const LOGOUT = 'LOGOUT';

export const SESSION_INFO_SET = 'SESSION_INFO_SET';
export const IS_LOGGED_IN_SET = 'IS_LOGGED_IN_SET';
export const ACTIVE_CHAT_SET = 'ACTIVE_CHAT_SET';
export const DRAWER_IS_OPEN_SET = 'DRAWER_IS_OPEN_SET';
export const CHANGE_ACTIVE_CHAT = 'CHANGE_ACTIVE_CHAT';
export const ADD_CHAT_DIALOG_IS_OPEN_SET = 'ADD_CHAT_DIALOG_IS_OPEN_SET';
export const MODIFY_CHAT_DIALOG_IS_OPEN_SET = 'MODIFY_CHAT_DIALOG_IS_OPEN_SET';
export const USER_MODIFY_DIALOG_IS_OPEN_SET = 'USER_MODIFY_DIALOG_IS_OPEN_SET';
