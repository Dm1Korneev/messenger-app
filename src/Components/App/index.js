import commonHoc from 'Components/commonHoc';

import { loginFromStore } from 'Redux/actions';
import { chatDialogIsOpenSelector, isLoggedInSelector, userModifyDialogIsOpenSelector } from 'Selectors/session';

import App from './App';

const mapStateToProps = (state) => ({
  chatDialogIsOpen: chatDialogIsOpenSelector(state),
  userModifyDialogIsOpen: userModifyDialogIsOpenSelector(state),
  isLoggedIn: isLoggedInSelector(state),
});

const mapDispatchToProps = {
  loginFromStore,
};

const Container = commonHoc(App, {
  mapStateToProps,
  mapDispatchToProps,
});

export { Container as App };
