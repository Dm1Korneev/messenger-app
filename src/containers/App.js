import { connect } from "react-redux";
import App from "../components/App";
import { loginFromStore } from "../redux/actions";

const mapStateToProps = state => ({
  chatDialogIsOpen:
    state.session.addChatDialogIsOpen || state.session.modifyChatDialogIsOpen,
  userModifyDialogIsOpen: state.session.userModifyDialogIsOpen,
  isLoggedIn: state.session.isLoggedIn
});

const mapDispatchToProps = dispatch => ({
  loginFromStore: () => dispatch(loginFromStore())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
