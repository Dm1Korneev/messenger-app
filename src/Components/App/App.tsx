import Box from '@material-ui/core/Box';
import CssBaseline from '@material-ui/core/CssBaseline';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';

import { MessengerScreen } from 'Components/MessengerScreen';
import { SignIn } from 'Components/SignIn';
import { useIsLoggedIn } from 'Hooks';
import { loginFromStore } from 'Redux/actions';

export const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loginFromStore());
  }, [dispatch]);

  const isLoggedIn = useIsLoggedIn();

  return (
    <Box height="100vh" display="flex">
      <CssBaseline />
      {isLoggedIn ? <MessengerScreen /> : <SignIn />}
    </Box>
  );
};
