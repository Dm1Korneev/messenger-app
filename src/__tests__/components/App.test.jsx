import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import App from 'Components/App';

const emptyStore = createStore(() => {});

jest.mock('Containers/TopBar', () => 'TopBar');
jest.mock('Containers/SignIn', () => 'SignIn');
jest.mock('Containers/SideBar', () => 'SideBar');
jest.mock('Components/MainContent', () => 'MainContent');
jest.mock('@material-ui/core/CssBaseline', () => 'CssBaseline');

describe('render app when not logged', () => {
  let wrapper;

  beforeAll(() => {
    const props = {
      loginFromStore: jest.fn(),
      chatDialogIsOpen: false,
      userModifyDialogIsOpen: false,
      isLoggedIn: false,
    };
    wrapper = global.mount(
      <Provider store={emptyStore}>
        <App {...props} />
      </Provider>,
    );
  });

  test('component is render', () => {
    expect(wrapper.find('App').length).toBe(1);
  });

  test('SignIn subcomponent is render', () => {
    wrapper.update();
    expect(wrapper.find('SignIn').length).toBe(1);
  });
});

describe.only('render app when logged', () => {
  let wrapper;

  beforeAll(() => {
    const props = {
      loginFromStore: jest.fn(),
      chatDialogIsOpen: false,
      userModifyDialogIsOpen: false,
      isLoggedIn: true,
    };
    wrapper = global.mount(
      <Provider store={emptyStore}>
        <App {...props} />
      </Provider>,
    );
  });

  test.only('MainContent subcomponent is render', () => {
    wrapper.update();
    // console.log(wrapper2.debug());
    expect(wrapper.dive().find('MainContent').length).toBe(1);
  });
});

describe('snapshot-test App component', () => {
  test('Renders correct properties', () => {
    global.shallowExpect(
      <Provider store={global.store}>
        <App />
      </Provider>,
    ).toMatchSnapshot();
  });
});
