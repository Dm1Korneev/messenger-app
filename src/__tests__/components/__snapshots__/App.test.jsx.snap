// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot-test App component Renders correct properties 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "chats": Object {
          "allIds": Array [],
          "byId": Object {},
        },
        "errors": Object {
          "LOGIN_FROM_STORE": undefined,
        },
        "loading": Object {
          "LOGIN_FROM_STORE": false,
        },
        "messages": Object {
          "allIds": Array [],
          "byChats": Object {},
          "byId": Object {},
        },
        "session": Object {
          "activeChat": undefined,
          "addChatDialogIsOpen": false,
          "drawerIsOpen": true,
          "isLoggedIn": false,
          "modifiableChatId": undefined,
          "modifyChatDialogIsOpen": false,
          "token": undefined,
          "user": undefined,
          "userModifyDialogIsOpen": false,
        },
        "users": Object {
          "allIds": Array [],
          "byId": Object {},
        },
      },
    }
  }
>
  <Connect(WithStyles(App)) />
</ContextProvider>
`;
